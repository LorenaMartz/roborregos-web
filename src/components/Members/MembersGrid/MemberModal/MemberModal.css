.row {
	margin: 0;
	padding: 0;
}

.member-modal-container {
	width: 100%;
	height: 100%;
	margin: 0;
	padding: 0;
	background: #000;
}

.container-helper {
	width: 100%;
	height: 100%;
	background:
		linear-gradient(
			90deg,
			rgba(19, 25, 30, 0.8) 40%,
			rgba(10, 15, 20, 0.8) 100%
		),
		url(images/github_pattern_white.svg) top left repeat fixed;
}

.main-modal-row {
	height: 100%;
}

.image-col {
	display: flex;
  align-items: center;
}

.image-cropper {
	width: 304px;
	height: 304px;
  overflow: hidden;
  border-radius: 50%;
}

.modal-member-image {
	height: 100%;
	transform: translateX(-16%);
}

.information-col {
	background: rgb(10, 15, 20);
}

@media (min-width: 1025px) {
	.information-col {
		height: 100%;
		display: flex;
	  align-items: center;
	}
}

@media (min-width: 320px) and (max-width: 1024px) {
	.information-col {
		width: 100%;
	}
}

.information-col .information-container {
	padding: 10px;
}

.information-col h2 {
	color: rgb(20, 183, 191);
}

.information-col p {
	color: rgba(255, 255, 255, 0.9);
	margin: 0;
}

.information-col p strong {
	font-size: 20px;
}

.information-col .member-modal-btn {
	color: rgba(255, 255, 255, 0.9);
	font-size: 18px;
	text-transform: none;
}

.information-col .member-modal-btn .member-username {
	padding-left: 8px;
}