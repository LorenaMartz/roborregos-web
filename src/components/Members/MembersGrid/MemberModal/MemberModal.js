import React, {Component} from 'react';
import {Row, Col} from 'react-bootstrap';
import Button from '@material-ui/core/Button';
import CloseIcon from '@material-ui/icons/Close';
import Icon from '@material-ui/core/Icon';
import IconButton from '@material-ui/core/IconButton';
import placeholder from '../../../../images/placeholder-rectangle.png';
import {MEDIUM_WIDTH} from '../../../../constants.js';
import './MemberModal.css';

/** Component class of Members' grid. */
class MemberModal extends Component {

  /**
  * Class constructor
  * @param {active_members} props: Lists of active and inactive members.
  */
  constructor(props) {
    super(props)

    this.tryRequire = this.tryRequire.bind(this);
    this.memberFullName = this.memberFullName.bind(this);
    this.largeView = this.largeView.bind(this);
    this.smallView = this.smallView.bind(this);
    this.updateSizeView = this.updateSizeView.bind(this);
    this.getContactButton = this.getContactButton.bind(this);
    this.generateDataButtons = this.generateDataButtons.bind(this);

    this.member = props.member;
    this.handleHideModal = props.onHide;

    this.state = {
      show_large: (window.innerWidth >= MEDIUM_WIDTH) ? true : false,
    };
  }

  /** Adds event listeners for modal hiding and responsivity. */
  componentDidMount() {
    document.addEventListener('keydown', this.escFunction, false)
    window.addEventListener('resize', this.updateSizeView)
  }

  /**
  * Parses path to member image
  * @param {string} imgPath: Path to member's photographs.
  * @return {path}
  */
  tryRequire(imgPath) {
    try {
      return require('images/members/' + imgPath);
    } catch (err) {
      return placeholder
    }
  }

  /**
  * Parses mamber's full name
  * @return {string}
  */
  memberFullName() {
    return `${this.member.name} ${this.member.lastname}`
  }

  /** Updates state.show_large for responsivity. */
  updateSizeView() {
    this.setState({
      show_large: (window.innerWidth >= MEDIUM_WIDTH),
    })
  }


  /**
  * Parses contact button links for each member.
  * @param {string} platform: Social media platform.
  * @param {string} className: String for classname.
  * @return {Element}
  */
  getContactButton(platform, className) {
    let href; let icon; let user;

    switch (platform) {
      case 'github':
        href = this.member.github
        icon = 'fa-github'
        user = this.member.github_user
        break
      case 'linkedin':
        href = this.member.linkedin
        icon = 'fa-linkedin'
        user = 'LinkedIn'
        break
      case 'resume':
        href = this.member.resume_link
        icon = 'fa-file-pdf'
        user = 'Resume'
        break
      default:
        break
    }
    return (
      <Button href={ href } className={ className }>
        <Icon className={ `fab ${ icon } fa-fw` } />
        <span className='member-username'>
          { user }
        </span>
      </Button>
    );
  }

  /**
  * Generates buttons for member's contact info.
  * @return {Element} Button
  */
  generateDataButtons() {
    let className = '';
    if (this.state.show_large) {
      className = 'member-modal-btn';
    } else {
      className = 'icon-small';
    }
    return (
      <div className='data-buttons'>
        { (this.member.github !== '') ?
          this.getContactButton('github', className) : null }
        <br/>
        { (this.member.linkedin !== '') ?
          this.getContactButton('linkedin', className) : null }
        <br/>
        { (this.member.resume_link !== '') ?
          this.getContactButton('resume', className) : null }
      </div>
    )
  }

  /**
  * Generates large view responsive components
  * @return {Element} Mmember modal
  */
  largeView() {
    return (
      <div className="member-modal-container">
        <div className="container-helper">
          <Row className="justify-content-center main-modal-row">
            <Col lg="3" className="image-col">
              <div className="image-cropper">
                <img
                  className="modal-member-image"
                  src={this.tryRequire(`${this.member.id}.jpg`)}
                  alt={this.memberFullName()}
                />
              </div>
            </Col>
            <Col lg="5" className="information-col">
              <div className="information-container">
                <Row noGutters>
                  <Col xs={{span: 1, offset: 11}}>
                    <IconButton
                      className='closing-btn'
                      onClick={ this.handleHideModal }
                    >
                      <CloseIcon />
                    </IconButton>
                  </Col>
                </Row>
                <Row noGutters>
                  <div className="member-titles">
                    <h2>
                      { this.memberFullName() }
                    </h2>
                    <strong> { this.member.role } </strong>
                    <div
                      className='member-subtitles'
                      style={{display: (this.member.status === 'inactive') ?
                      'block' : 'none'}}
                    >
                      <h6>
                        <div> { this.member.subtitle } </div>
                        <div>
                          {'Since ' + this.member.class +
                          ', ' + this.member.semesters +
                          ' semesters' }
                        </div>
                      </h6>
                    </div>
                  </div>

                </Row>
                <Row noGutters>
                  <p className="member-data">
                    { this.member.description }
                  </p>
                </Row>
                <Row noGutters className="justify-content-center">
                  { this.generateDataButtons() }
                </Row>
              </div>
            </Col>
          </Row>
        </div>
      </div>
    )
  }

  /**
  * Generates small view responsive components
  * @return {Element} Mmember modal
  */
  smallView() {
    return (
      <div className="member-modal-container">
        <div className="container-helper">
          <Row className="main-modal-row">
            <div className="close-button">
              <Col>
                <IconButton
                  className='icon-small'
                  onClick={ this.handleHideModal}
                >
                  <CloseIcon />
                </IconButton>
              </Col>
            </div>
            <Row noGutters className="image-col-small">
              <div className="image-cropper">
                <img
                  className="modal-member-image"
                  src={this.tryRequire(`${this.member.id}.jpg`)}
                  alt={this.memberFullName()}
                />
              </div>
            </Row>
            <div className="description-small">
              <Row noGutters>
                <div className="member-titles">
                  <h2 className="name-small">
                    { this.memberFullName() }
                  </h2>
                  <strong> { this.member.role } </strong>
                  <div
                    className='member-subtitles'
                    style={{display: (this.member.status === 'inactive') ?
                    'block' : 'none'}}>
                    <h6>
                      <div> { this.member.subtitle } </div>
                      <div>
                        {'Since ' + this.member.class + ', ' +
                        this.member.semesters + ' semesters' }
                      </div>
                    </h6>
                  </div>
                </div>
              </Row>
              <Row noGutters>
                <div className="member-data">
                  { this.member.description }
                </div>
              </Row>
              <Row noGutters className="justify-content-center">
                { this.generateDataButtons() }
              </Row>
            </div>
          </Row>
        </div>
      </div>
    )
  }

  /**
  * Render funciton of member modal
  * @return {components} Memmebr modal
  */
  render() {
    if (this.state.show_large) {
      return this.largeView()
    }
    return this.smallView()
  }
}

export default MemberModal;
