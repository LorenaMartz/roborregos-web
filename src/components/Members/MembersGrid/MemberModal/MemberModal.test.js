// @flow
import { getByTestId } from '@testing-library/react'
import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import MemberModal from './MemberModal'
// Mock data
const member = {
  id: 1,
  name: 'First',
  lastname: 'Member1',
  status: 'inactive',
  role: 'Software Development',
  class: '2015',
  semesters: '8',
  degree: 'ISD',
  description: 'Mock member #1.',
  github: 'https://github.com/1',
  github_user: 'git_username1',
  linkedin: 'https://www.linkedin.com/1',
  resume_link: '',
  tags: '',
  Email: 'mock1@gmail.com',
}

let container = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  if (document.body != null) {
    document.body.appendChild(container)
  } else {
    expect(document.body).not.toEqual(null)
  }
})

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  if (container != null) {
    container.remove()
  } else {
    expect(container).not.toEqual(null)
  }
  container = null
})

it('<Member Modal> Renders correctly (large view)', () => {
  act(() => {
    render(<MemberModal
      member={member}
      onHide={() => {}}
    />, container)
  })
  expect(getByTestId(container, 'member-modal-container')).not.toEqual(null)
  expect(getByTestId(container, 'information-container')).not.toEqual(null)
  expect(getByTestId(container, 'member-data')).not.toEqual(null)
})

// it('<Member Modal> Closes', () => {
//   act(() => {
//     render(<MemberModal
//       member={members[0]}
//       onHide={this.handleHideModal}
//     />, container)
//   })
//   expect(getByTestId(container, 'member-modal-container')).not.toEqual(null)
//   expect(getByTestId(container, 'information-container')).not.toEqual(null)
//   expect(getByTestId(container, 'member-data')).not.toEqual(null)
// })

it('<Member Modal> has correct data', () => {
  act(() => {
    render(<MemberModal
      member={member}
      onHide={() => {}}
    />, container)
  })
  expect(getByTestId(container, 'member-data')).not.toEqual(null)
  expect(getByTestId(container, 'member-data').textContent).toBe(member.description)
  expect(getByTestId(container, 'member-fullName')).not.toEqual(null)
  expect(getByTestId(container, 'member-fullName').textContent).toBe(`${member.name} ${member.lastname}`)
  expect(getByTestId(container, 'member-subtitles')).not.toEqual(null)
  expect(getByTestId(container, 'member-subtitles').textContent)
    .toBe(`Since ${member.class}, ${member.semesters} semesters`)
  expect(getByTestId(container, member.github_user)).not.toEqual(null)
  expect(getByTestId(container, member.github_user).textContent).toBe(member.github_user)
  expect(getByTestId(container, 'LinkedIn')).not.toEqual(null)
//   expect(getByTestId(container, 'LinkedIn').textContent).toBe(member.LinkedIn)
//   expect(getByTestId(container, 'Resume')).toEqual(null)
})

it('Matches snapshot large view', () => {
  const tree = renderer.create(
    <MemberModal
      member={member}
      onHide={() => {}}
    />,
  ).toJSON()
  expect(tree).toMatchSnapshot('MemberModalLargeView')
})
