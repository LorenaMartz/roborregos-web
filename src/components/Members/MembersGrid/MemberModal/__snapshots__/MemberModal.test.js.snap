// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot large view: MemberModalLargeView 1`] = `
<div
  className="member-modal-container"
  data-testid="member-modal-container"
>
  <div
    className="container-helper"
  >
    <div
      className="justify-content-center main-modal-row row"
    >
      <div
        className="image-col col-lg-3"
      >
        <div
          className="image-cropper"
        >
          <img
            alt="First Member1"
            className="modal-member-image"
            src="placeholder-rectangle.png"
          />
        </div>
      </div>
      <div
        className="information-col col-lg-5"
      >
        <div
          className="information-container"
          data-testid="information-container"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-1 offset-11"
            >
              <button
                className="MuiButtonBase-root MuiIconButton-root closing-btn"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            className="row no-gutters"
          >
            <div
              className="member-titles"
            >
              <h2
                data-testid="member-fullName"
              >
                First Member1
              </h2>
              <strong>
                 
                Software Development
                 
              </strong>
              <div
                className="member-subtitles"
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <h6>
                  <div>
                     
                     
                  </div>
                  <div
                    data-testid="member-subtitles"
                  >
                    Since 2015, 8 semesters
                  </div>
                </h6>
              </div>
            </div>
          </div>
          <div
            className="row no-gutters"
          >
            <p
              className="member-data"
              data-testid="member-data"
            >
              Mock member #1.
            </p>
          </div>
          <div
            className="justify-content-center row no-gutters"
          >
            <div
              className="data-buttons"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-text member-modal-btn"
                href="https://github.com/1"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    aria-hidden={true}
                    className="material-icons MuiIcon-root fab fa-github fa-fw"
                  />
                  <span
                    className="member-username"
                    data-testid="git_username1"
                  >
                    git_username1
                  </span>
                </span>
              </a>
              <br />
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-text member-modal-btn"
                href="https://www.linkedin.com/1"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    aria-hidden={true}
                    className="material-icons MuiIcon-root fab fa-linkedin fa-fw"
                  />
                  <span
                    className="member-username"
                    data-testid="LinkedIn"
                  >
                    LinkedIn
                  </span>
                </span>
              </a>
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
